### DISCUSSION TITLE
NextJS Plugin

### DISCUSSION THREAD

**Discussion by zwgnr** [macaron-css/macaron#16] [Ideas]
Great work on this library so far!

Is there a plugin for NextJS in the works? If not, I may be interested in getting to work on one!

**Comment by kavinvalli** [comment]
> Great work on this library so far!
> 
> Is there a plugin for NextJS in the works? If not, I may be interested in getting to work on one!

Hey @zwagnr. As far as I know, to get Macaron working with NextJS, we need to get the webpack plugin up and running. The webpack plugin seems to be under development:
https://macaron.js.org/docs/examples/webpack/

@Mokshit06 will probably be the right person to give you an update on this, though

**Comment by Mokshit06** [comment]
Hey @zwagnr, as @kavinvalli said, macaron doesn't have a nextjs plugin yet.
I did start making a webpack plugin, but it doesn't do much right now since I was trying out webpack's API, so it would probably be better to start from scratch

If you're interested in working on it that's great! I can help you with understanding macaron's internals, or you can see the implementation of other plugins

**Comment by Gregoor** [comment]
Did someone already start working on this? I'm wondering what the best forward would be, since webpack's time within NextJS seems to be running out, while [this part of the docs](https://turbo.build/pack/docs/features/customizing-turbopack#webpack-loaders-for-nextjs) makes me think that turbopack would not yet support this kind of loader (the one that generates CSS files).

That said, Webpack probably won't fade out before Turbopack supports such features, thus developing a webpack loader might still be the way to go for now?
